runner/local/simple.go

```diff
@@ -32,26 +39,33 @@ func (r *simpleRunner) Run(cmd *runner.Command) runner.ProcessStatus {
 	runId := runner.RunId(fmt.Sprintf("%d", r.nextRunId))
 	r.nextRunId++
 
-	if len(r.running) == 1 {
+	if r.running != nil {
 		r.runs[runId] = runner.BadRequestStatus(runId, fmt.Errorf("Runner is busy"))
 		return r.runs[runId]
 	}
 
-	var c execer.Command
-	c.Argv = cmd.Argv
-	p, err := r.exec.Exec(c)
-	if err != nil {
-		r.runs[runId] = runner.BadRequestStatus(runId, fmt.Errorf("could not exec: %v", err))
-		return r.runs[runId]
-	}
-	r.runs[runId] = runner.RunningStatus(runId)
-	r.running[runId] = p
-
-	go babysit(p, runId, r, cmd.Timeout)
+	r.running = &inflight{runId: runId, cancelCh: make(chan struct{})}
+	r.runs[runId] = runner.PreparingStatus(runId)
 
+	// Run in a new goroutine
+	go r.run(cmd, runId, r.running.cancelCh)
+	if cmd.Timeout > 0 { // Timeout if applicable
+		time.AfterFunc(cmd.Timeout, func() { r.updateStatus(runner.TimeoutStatus(runId)) })
```

Comment:
It hints at how often we abort/timeout a the last moment. Again, not a great example.